### PR TITLE
Merge `php8` back into `master`

### PR DESCRIPTION
**Pull request by czosel**
Updated version of #807 with resolved conflicts

### CODE REVIEW

File: src/lexer/scripting.js
Comment (czosel) [reply src/lexer/scripting.js]: fixed

File: src/parser/try.js
Comment (czosel) [reply src/parser/try.js:31]: fixed

File: test/snapshot/attributes.test.js
Comment (czosel) [reply test/snapshot/attributes.test.js:159]: Good call - turns out we're currently dropping the first line instead of parsing it as a comment. I'm not sure about how to fix this correctly, shall we create a separate issue for it?

File: test/snapshot/class.test.js
Comment (czosel) [reply test/snapshot/class.test.js:191]: This case is already covered here: https://github.com/glayzzle/php-parser/pull/809/files#diff-afaacbd57a577ff9abf9ab4d44fc4ca9a4a814f5ed5fb7a6b381494d268bb5c0R32

File: test/snapshot/attributes.test.js
Comment (czosel) [reply test/snapshot/attributes.test.js:128]: added a test for anonymous classes and functions

**Comment by czosel** [comment]
Hi @ichiriac :slightly_smiling_face:  Do you think we can move forward with this? The only open issue is https://github.com/glayzzle/php-parser/pull/809#discussion_r739681258, which we can address in a separate PR i think.

File: test/snapshot/attributes.test.js
Comment (cseufert) [reply test/snapshot/attributes.test.js:182]: Yes, I guess we need to check if the attribute was consumed, and if not, throw an error. Sorry been absolutely flat out, but will have a look when I get some time.

Comment (czosel) [reply test/snapshot/attributes.test.js:182]: @cseufert no worries, thanks for the feedback!

**Comment by czosel** [comment]
@ichiriac friendly bump 🙃

**Comment by czosel** [comment]
Hi @ichiriac, happy holidays :slightly_smiling_face: Is there anything I can do to help move this forward?

**Comment by jebarjonet** [comment]
@czosel Thanks for this PR 🙂 Should you find someone else to review this PR since @ichiriac has not been answering for months ? This is a pretty big update and it will be useful to a couple of linting and formatting tools for PHP 8.

**Comment by jaulz** [comment]
Maybe @b4dnewz can help since he is the second member of glayzzle?

**Comment by czosel** [comment]
@jebarjonet Yeah, it would be great if someone else could take a look. I'm not sure if anyone else next to @ichiriac knows the codebase well enough, but I guess slowly the benefits of moving this forward outweigh the risk of introducing issues (which can then be fixed). I also have merge permissions on the repo, but I'd prefer not to use them without anyone else's approval :wink:

**Comment by alexander-akait** [comment]
@ichiriac Friendly ping

**Review by alexander-akait — APPROVED** [review]

**Pull request merged**
